hive/src/main/java/io/confluent/connect/storage/hive/schema/DefaultSchemaGenerator.java

```diff
@@ -36,9 +36,16 @@ public DefaultSchemaGenerator(Map<String, Object> config) {
   }
 
   @Override
-  public List<FieldSchema> newPartitionFields(String partitionField) {
-    return Collections.singletonList(
-        new FieldSchema(partitionField, TypeInfoFactory.stringTypeInfo.toString(), "")
-    );
+  public List<FieldSchema> newPartitionFields(String format) {
+    String[] fields = format.split(",");
+    final List<FieldSchema> result = new ArrayList<>();
```

Comment:
nit: Access pattern of result doesn't differ from that of `fields`. They are both assigned once. You may remove `final` here to make this clear.